src/wmtk/simplex/closed_star.cpp

```diff
@@ -0,0 +1,35 @@
+#include "closed_star.hpp"
+
+#include <wmtk/TetMesh.hpp>
+#include <wmtk/TriMesh.hpp>
+
+#include "coface_cells.hpp"
+#include "simplex_boundary.hpp"
+
+namespace wmtk::simplex {
+
+SimplexCollection closed_star(const Mesh& mesh, const Simplex& simplex, const bool sort_and_clean)
+{
+    SimplexCollection collection(mesh);
+
+    collection.add(simplex);
+
+    const SimplexCollection coface_cells_collection =
+        mesh.top_simplex_type() == PrimitiveType::Face
+            ? coface_cells(static_cast<const TriMesh&>(mesh), simplex, false)
+            : coface_cells(static_cast<const TetMesh&>(mesh), simplex, false);
+
+    for (const Simplex& coface_cell : coface_cells_collection.simplex_vector()) {
+        collection.add(coface_cell);
```

Comment:
if we go with my thought of making add a sorted insertion (which results in O(n) insertion) it's probably better to create a `std::vector<Simplex>` to accumulate new simplices, sort it, and then merge.

Reply:
Once more, I am not sure if we want to keep things sorted at all times. We need to discuss it.